docs/source/examples/best-practices.rst

```diff
@@ -0,0 +1,112 @@
+Best Practices
+==============
+
+
+Multi-GPUs
+~~~~~~~~~~
+
+When using multiple GPUs, if possible, it's best to co-locate as many GPUs on the same physical node.  This could be a
+DGX, a cloud instance with `multi-gpu options <https://rapids.ai/cloud>`_ , a high-density GPU HPC instance, etc.  This is done for
+two reasons:
+
+- 1. Moving data between GPUs is costly and performance decreases when computation stops due to communication overheads, Host-to-Device/Device-to-Host transfers, etc
+- 2. Mutli-GPU instances often come with accelerated networking like `NVLink <https://www.nvidia.com/en-us/data-center/nvlink/>`_.  These accelerated
+networking paths usually have much higher throughput/bandwidth compared with traditional networking *and* don't force and H-to-D/D-to-H transfers.  See `
+Accelerated Networking`_ for more discussion
+
+.. code-block:: python
+
+    from dask_cuda import LocalCUDACluster
+
+    cluster = LocalCUDACluster(CUDA_VISIBLE_DEVICES="0,1",
+                               protocol="ucx")
+
+GPU Memory Management
+~~~~~~~~~~~~~~~~~~~~~
+
+When using Dask-CUDA, especially with RAPIDS, it's best to use an `RMM pool <https://docs.rapids.ai/api/rmm/stable/>`_ (RAPIDS Memory Manager)
+to pre-allocate memory on the GPU.  Allocating memory, while fast, takes a small amount of time, however, one can easily make
+hundreds of thousand or even millions of allocations in trivial workflows causing significant performance degradations.  With an RMM pool, allocations are sub-sampled from a larger
+pool and greatly reduces the allocation time and thereby increases performance:
```

Comment:
This also has a big positive effect when using UCX with cuda IPC: we only need to register a single IPC handle for the whole pool (which is expensive, but only done once) since from the IPC point of viewer there's only a single allocation. As opposed to just using RMM without a pool where each new allocation must be registered with IPC.